File: VB/SpreadsheetChartAPISamples/CodeExamples/ViewOptionsActions.vb
Imports System
Imports System.Drawing
Imports System.Globalization
Imports DevExpress.Spreadsheet
Imports DevExpress.Spreadsheet.Charts
Imports DevExpress.Spreadsheet.Drawings
Imports DevExpress.Utils

Namespace SpreadsheetChartAPIActions

    Public Module ViewOptionsActions

        Private Sub ShowAutomaticMarkers(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
'#Region "#ShowAutomaticMarkers"
            Dim worksheet As DevExpress.Spreadsheet.Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet
            ' Create a chart and specify its location.
            Dim chart As DevExpress.Spreadsheet.Charts.Chart = worksheet.Charts.Add(DevExpress.Spreadsheet.Charts.ChartType.Line, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L15")
            ' Display markers using automatic style.
            chart.Series(CInt((0))).Marker.Symbol = DevExpress.Spreadsheet.Charts.MarkerStyle.Auto
            ' Hide the legend.
            chart.Legend.Visible = False
'#End Region  ' #ShowAutomaticMarkers
        End Sub

        Private Sub ShowCustomMarkers(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
'#Region "#ShowCustomMarkers"
            Dim worksheet As DevExpress.Spreadsheet.Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet
            ' Create a chart and specify its location.
            Dim chart As DevExpress.Spreadsheet.Charts.Chart = worksheet.Charts.Add(DevExpress.Spreadsheet.Charts.ChartType.Line, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L15")
            ' Display markers and specify the marker style.
            chart.Series(CInt((0))).Marker.Symbol = DevExpress.Spreadsheet.Charts.MarkerStyle.Circle
            ' Hide the legend.
            chart.Legend.Visible = False
'#End Region  ' #ShowCustomMarkers
        End Sub

        Private Sub SetMarkerSize(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
'#Region "#SetMarkerSize"
            Dim worksheet As DevExpress.Spreadsheet.Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet
            ' Create a chart and specify its location.
            Dim chart As DevExpress.Spreadsheet.Charts.Chart = worksheet.Charts.Add(DevExpress.Spreadsheet.Charts.ChartType.Line, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L15")
            ' Display markers and specify the marker style and size.
            chart.Series(CInt((0))).Marker.Symbol = DevExpress.Spreadsheet.Charts.MarkerStyle.Circle
            chart.Series(CInt((0))).Marker.Size = 15
            ' Hide the legend.
            chart.Legend.Visible = False
'#End Region  ' #SetMarkerSize
        End Sub

        Private Sub SmoothLines(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
'#Region "#SmoothLines"
            Dim worksheet As DevExpress.Spreadsheet.Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet
            ' Create a chart and specify its location.
            Dim chart As DevExpress.Spreadsheet.Charts.Chart = worksheet.Charts.Add(DevExpress.Spreadsheet.Charts.ChartType.LineMarker, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L15")
            ' Turn on curve smoothing.
            chart.Series(CInt((0))).Smooth = True
            ' Hide the legend.
            chart.Legend.Visible = False
'#End Region  ' #SmoothLines
        End Sub

        Private Sub GapWidth(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
'#Region "#GapWidth"
            Dim worksheet As DevExpress.Spreadsheet.Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet
            ' Create a chart and specify its location.
            Dim chart As DevExpress.Spreadsheet.Charts.Chart = worksheet.Charts.Add(DevExpress.Spreadsheet.Charts.ChartType.ColumnClustered, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L15")
            ' Set the gap width between data series.
            chart.Views(CInt((0))).GapWidth = 33
            ' Hide the legend.
            chart.Legend.Visible = False
'#End Region  ' #GapWidth
        End Sub

        Private Sub VaryColorsByPoint(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
'#Region "#VaryColorsByPoint"
            Dim worksheet As DevExpress.Spreadsheet.Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet
            ' Create a chart and specify its location.
            Dim chart As DevExpress.Spreadsheet.Charts.Chart = worksheet.Charts.Add(DevExpress.Spreadsheet.Charts.ChartType.ColumnClustered, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L15")
            ' Specify that each data point in the series has a different color.
            chart.Views(CInt((0))).VaryColors = True
            ' Hide the legend.
            chart.Legend.Visible = False
'#End Region  ' #VaryColorsByPoint
        End Sub
    End Module
End Namespace

File: CS/SpreadsheetChartAPISamples/CodeExamples/ViewOptionsActions.cs
using System;
using System.Drawing;
using System.Globalization;
using DevExpress.Spreadsheet;
using DevExpress.Spreadsheet.Charts;
using DevExpress.Spreadsheet.Drawings;
using DevExpress.Utils;

namespace SpreadsheetChartAPIActions {
    public static class ViewOptionsActions {
        static void ShowAutomaticMarkers(IWorkbook workbook) {
            #region #ShowAutomaticMarkers
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location.
            Chart chart = worksheet.Charts.Add(ChartType.Line, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L15"];

            // Display markers using automatic style.
            chart.Series[0].Marker.Symbol = MarkerStyle.Auto;
            // Hide the legend.
            chart.Legend.Visible = false;

            #endregion #ShowAutomaticMarkers
        }

        static void ShowCustomMarkers(IWorkbook workbook) {
            #region #ShowCustomMarkers
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location.
            Chart chart = worksheet.Charts.Add(ChartType.Line, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L15"];

            // Display markers and specify the marker style.
            chart.Series[0].Marker.Symbol = MarkerStyle.Circle;
            // Hide the legend.
            chart.Legend.Visible = false;

            #endregion #ShowCustomMarkers
        }

        static void SetMarkerSize(IWorkbook workbook) {
            #region #SetMarkerSize
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location.
            Chart chart = worksheet.Charts.Add(ChartType.Line, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L15"];

            // Display markers and specify the marker style and size.
            chart.Series[0].Marker.Symbol = MarkerStyle.Circle;
            chart.Series[0].Marker.Size = 15;
            // Hide the legend.
            chart.Legend.Visible = false;

            #endregion #SetMarkerSize
        }

        static void SmoothLines(IWorkbook workbook) {
            #region #SmoothLines
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location.
            Chart chart = worksheet.Charts.Add(ChartType.LineMarker, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L15"];

            // Turn on curve smoothing.
            chart.Series[0].Smooth = true;
            // Hide the legend.
            chart.Legend.Visible = false;

            #endregion #SmoothLines
        }

        static void GapWidth(IWorkbook workbook) {
            #region #GapWidth
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location.
            Chart chart = worksheet.Charts.Add(ChartType.ColumnClustered, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L15"];

            // Set the gap width between data series.
            chart.Views[0].GapWidth = 33;
            // Hide the legend.
            chart.Legend.Visible = false;

            #endregion #GapWidth
        }

        static void VaryColorsByPoint(IWorkbook workbook) {
            #region #VaryColorsByPoint
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location.
            Chart chart = worksheet.Charts.Add(ChartType.ColumnClustered, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L15"];

            // Specify that each data point in the series has a different color.
            chart.Views[0].VaryColors = true;
            // Hide the legend.
            chart.Legend.Visible = false;

            #endregion #VaryColorsByPoint
        }

    }
}
